Cabal/Distribution/PackageDescription/Parsec.hs

```diff
@@ -125,7 +125,7 @@ parseGenericPackageDescription bs = do
     setCabalSpecVersion ver
     -- if we get too new version, fail right away
     case ver of
-        Just v | v > mkVersion [2,2] -> parseFailure zeroPos
+        Just v | v > mkVersion [2,4] -> parseFailure zeroPos
```

Comment:
Hmm, why did this work before? We don't have `.cabal` files with `cabal-version: 2.4` in the test suite?

Reply:
Apparently not! I only found I needed to do this when I tried to add one in the course of #5403.

Reply:
Nice, that means that with #5403 we will have one.